src/openzeppelin/access/ownable.cairo

```diff
@@ -5,48 +5,101 @@
 
 from starkware.cairo.common.cairo_builtins import HashBuiltin
 from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.math import assert_not_zero
+
+#
+# Events
+#
+
+@event
+func OwnershipTransferred(previousOwner: felt, newOwner: felt):
+end
+
+#
+# Storage
+#
 
 @storage_var
-func Ownable_owner() -> (owner: felt):
+func _owner() -> (owner: felt):
 end
 
+#
+# Constructor
+#
+
 func Ownable_initializer{
-        syscall_ptr : felt*, 
+        syscall_ptr : felt*,
         pedersen_ptr : HashBuiltin*,
         range_check_ptr
     }(owner: felt):
-    Ownable_owner.write(owner)
+    _transferOwnership(owner)
     return ()
 end
 
-func Ownable_only_owner{
-        syscall_ptr : felt*, 
+#
+# Protector (Modifier)
+#
+
+func Ownable_onlyOwner{
+        syscall_ptr : felt*,
         pedersen_ptr : HashBuiltin*,
         range_check_ptr
     }():
-    let (owner) = Ownable_owner.read()
+    let (owner) = Ownable_owner()
     let (caller) = get_caller_address()
     with_attr error_message("Ownable: caller is not the owner"):
         assert owner = caller
     end
     return ()
 end
 
-func Ownable_get_owner{
-        syscall_ptr : felt*, 
+#
+# Getters
+#
+
+func Ownable_owner{
+        syscall_ptr : felt*,
         pedersen_ptr : HashBuiltin*,
         range_check_ptr
     }() -> (owner: felt):
-    let (owner) = Ownable_owner.read()
+    let (owner) = _owner.read()
     return (owner=owner)
 end
 
-func Ownable_transfer_ownership{
-        syscall_ptr : felt*, 
+func Ownable_transferOwnership{
+        syscall_ptr : felt*,
         pedersen_ptr : HashBuiltin*,
         range_check_ptr
-    }(new_owner: felt) -> (new_owner: felt):
-    Ownable_only_owner()
-    Ownable_owner.write(new_owner)
-    return (new_owner=new_owner)
+    }(newOwner: felt):
+    with_attr error_message("Ownable: new owner is the zero address"):
+        assert_not_zero(newOwner)
+    end
+    Ownable_onlyOwner()
+    _transferOwnership(newOwner)
+    return ()
+end
+
+func Ownable_renounceOwnership{
+        syscall_ptr : felt*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }():
+    Ownable_onlyOwner()
+    _transferOwnership(0)
+    return ()
+end
+
+#
+# Internal
+#
+
+func _transferOwnership{
```

Comment:
there's no reason for internal functions to use camelCase

Reply:
While being internal, it is not "private" ... I know this difference doesn't really translate into Cairo, but still it's important.

In solidity, the public/external `transferOwnership` can only be called by the current owner. This prevents use cases where internal logic decides that some payment or some signature triggers an ownership change. IMO, contracts that use the ownable pattern might, in addition to exposing the other function as external, also use these internal ones.

Just like `ERC20:_mint`

I was feeling that, for these devs, name consistency would be a plus. Just like they are used to call `ERC20:_transfer`, `ERC20:_mint`, `Ownable:_transferOwnership` or `AccessControl:_setRoleAdmin` in Solidity ... they might want to find the same functions, with the same name in Cairo

Reply:
> if we're ok with exposing an unguarded version of transfer ownership then we should prefix it. and probably name it accordingly, to prevent misusage.

Any function that is internal (and not private) in the solidity code, is intended to be used in child contracts. This definitely includes `_transfer` and `_approve` (among others)

Should we have
- `ERC20_transfer` for the public one
- `ERC20__transfer` for the internal one?
or
- `ERC20_internal_transfer` for the internal one?

Same for modifiers, maybe we should do one of
```
ERC20_modifier_onlyOwner
ERC20_modifier_only_owner
ERC20_guard_onlyOwner
ERC20_guard_only_owner
```

Reply:
Trying not to fall in the Solidity proximity trap, I think we should be mindful of the building blocks we wish to export and name functions with that in mind. For example I wouldn't call anything a modifier, since there's no modifiers in Cairoland. What's the purpose/usecase of exposing the internal "unguarded" transfer to the user? Maybe i'd just call it `ERC20_unguarded_transfer`.

Reply:
I would also argue that the snake_case vs camelCase thing shows its limits:

```
@storage_var
func ERC20_total_supply() -> (total_supply: Uint256):
end

...

func ERC20_totalSupply{
        syscall_ptr : felt*,
        pedersen_ptr : HashBuiltin*,
        range_check_ptr
    }() -> (totalSupply: Uint256):
    let (totalSupply: Uint256) = ERC20_total_supply.read()
    return (totalSupply)
end
```

Here its convenient that the external function is standardized, and so you use camelCase ... otherwise you get a name conflict. This is why there are sometimes trailing _

```
@storage_var
func ERC20_name_() -> (name: felt):
end

@storage_var
func ERC20_symbol_() -> (symbol: felt):
end

@storage_var
func ERC20_decimals_() -> (decimals: felt):
end

@storage_var
func ERC20_total_supply() -> (total_supply: Uint256):
end
```

These suffix have no other reason then to avoid name conflict. IMO this is bad design, and another convention would be welcome.

Maybe `ERC20_storage_name`, or just `_name` since storage isn't (IMO) supposed to be exported/reused

Note: If only the language allowed `import openzeppelin.token.erc20.library as ERC20` and then `ERC20.transfer` / `ERC20._transfer` then that would be way cleaner

Reply:
I think we agree on most points 👍 

> IMO this is not "fall in the Solidity proximity trap", but rather reproducing the features that enable modularity of the contract.
I think it's fine to offer a consistent API across Solidity and Cairo contracts. It's calling functions "modifiers" or naming functions after the Solidity library what I call a proximity trap. It's easy to distinguish between `approve` and `_approve` when you have inheritance and visibility rules, but we don't have that in Cairo. I think we should have similar building blocks but adapted to the dev environment.

This is calling them `ERC20_unguarded_approve` and `ERC20_unguarded_burn`, etc; while reserving camelCase to functions that are meant to be made external as-is, like `transferFrom`, and assuming single-word ones are too (`transfer`).

> Note: If only the language allowed import openzeppelin.token.erc20.library as ERC20 and then ERC20.transfer / ERC20._transfer then that would be way cleaner

Yes, we've raised this with the Cairo team before

Reply:
> Maybe ERC20_storage_name, or just _name since storage isn't (IMO) supposed to be exported/reused

I'm open to discuss whether storage should be encapsulated by library modules.

Reply:
Continued conversation in https://github.com/OpenZeppelin/cairo-contracts/discussions/267